influxdb3_write/src/lib.rs

```diff
@@ -84,6 +85,9 @@ pub trait Bufferer: Debug + Send + Sync + 'static {
 
     /// Returns the catalog
     fn catalog(&self) -> Arc<catalog::Catalog>;
+
+    /// Returns the queryable buffer
+    fn queryable_buffer(&self) -> Arc<QueryableBuffer>;
```

Comment:
Rather than giving access directly to the queryable buffer, can you put a function on this trait to get access to the parquet files? I think this will work more cleanly for when we have multiple queryable buffers in Pro and it will implement this trait to hide that fact.

Reply:
Yep, that makes sense - I'll give it a shot!

Reply:
I addressed this in https://github.com/influxdata/influxdb/pull/25225/commits/ff6e9a41415bbf59aca312722d89a756e8bb79ce by giving a `Arc<dyn WriteBuffer>` to the system table provider, which now has a method that produces parquet files for a given db/table.

Some `Arc` 🔥 was involved, which spread throughout the codebase.

---

One other thing I could think would be to produce an error if an invalid table name is provided. The underlying function on the `QueryableBuffer` that gives the parquet files is called in the query path where the db/table name are validated with the catalog. So, if an invalid table name is passed, it will produce an empty record set, vs. an error.

I can work that into this PR or create a follow-on issue to address that.

Reply:
Maybe save for a follow on

Reply:
I opened https://github.com/influxdata/influxdb/issues/25229 to track.